Entities/Characters/Knight/KnightLogic.as

```diff
@@ -1439,12 +1440,25 @@ void onCollision(CBlob@ this, CBlob@ blob, bool solid, Vec2f normal, Vec2f point
 				//printf("shi " + shieldVars.direction * normal );
 				if (shieldVars.direction * normal < 0.0f)
 				{
+					//print("" + vellen);
 					knight_add_actor_limit(this, blob);
 					this.server_Hit(blob, pos, vel, 0.0f, Hitters::shield);
 
 					Vec2f force = Vec2f(shieldVars.direction.x * this.getMass(), -this.getMass()) * 3.0f;
 
+					vellen = Maths::Min(vellen, 8.0f);
+
+					if (vellen < 3.5f)
+					{
+						force *= Maths::Pow(vellen, 1.0f / 3.0f) / 2;
+					}
+					else
+					{
+						force *= (vellen - 3.5f) / 6 + 0.759f;
+					}
```

Comment:
It probably would be good to include a comment explaining what this all means for someone looking at this in the future.

Reply:
done